src/app/state/hooks.js

```diff
@@ -221,6 +222,99 @@ export function useCoordination(parameters, coordinationScopes) {
   return [values, setters];
 }
 
+/**
+ * Get a mapping from dataset coordination scopes to dataset UIDs.
+ * @param {object} coordinationScopes The coordination scope mapping object for a view.
+ * @returns {object} Mapping from dataset coordination scope names to dataset UIDs.
+ */
+export function useDatasetUids(coordinationScopes) {
+  const parameter = CoordinationType.DATASET;
+  const datasetScopes = coordinationScopes[parameter];
+
+  // Mapping from dataset coordination scope name to dataset uid
+  const datasetUids = useViewConfigStore((state) => {
+    const { coordinationSpace } = state.viewConfig;
+    // Convert a single scope to an array of scopes to be consistent.
+    const datasetScopesArr = Array.isArray(datasetScopes) ? datasetScopes : [datasetScopes];
+    return fromEntries(datasetScopesArr.map((datasetScope) => {
+      if (coordinationSpace && coordinationSpace[parameter]) {
+        const value = coordinationSpace[parameter][datasetScope];
+        return [datasetScope, value];
+      }
+      return [datasetScope, undefined];
+    }));
+  }, shallow);
+
+  return datasetUids;
+}
+
+/**
+ * Use coordination values and coordination setter functions corresponding to
+ * dataset-specific coordination scopes for each coordination type.
+ * @param {string[]} parameters An array of coordination types supported by a view.
+ * @param {object} coordinationScopes The coordination scope mapping object for a view.
+ * @returns {array} [cValues, cSetters] where
+ * cValues is a mapping from coordination scope name to { coordinationType: coordinationValue },
+ * and cSetters is a mapping from coordination scope name to { setCoordinationType }
+ * setter functions.
+ */
+export function useMultiDatasetCoordination(parameters, coordinationScopes) {
+  const setCoordinationValue = useViewConfigStore(state => state.setCoordinationValue);
+
+  const datasetScopes = coordinationScopes[CoordinationType.DATASET];
+
+  // Convert a single scope to an array of scopes to be consistent.
+  const datasetScopesArr = Array.isArray(datasetScopes) ? datasetScopes : [datasetScopes];
+
+  const values = useViewConfigStore((state) => {
+    const { coordinationSpace } = state.viewConfig;
+    return fromEntries(datasetScopesArr.map((datasetScope) => {
+      const datasetValues = fromEntries(parameters.map((parameter) => {
+        if (coordinationSpace && coordinationSpace[parameter]) {
+          let value;
+          if (typeof coordinationScopes[parameter] === 'object') {
```

Comment:
Maybe assign `coordinationScopes[parameter]` to a variable to prevent code duplication